zipline/data/data_portal.py

```diff
@@ -463,41 +463,76 @@ def get_spot_value(self, asset, field, dt, data_frequency):
             ``field`` is 'volume' the value will be a int. If the ``field`` is
             'last_traded' the value will be a Timestamp.
         """
-        if self._is_extra_source(asset, field, self._augmented_sources_map):
-            return self._get_fetcher_value(asset, field, dt)
+        assets_is_scalar = False
+        if isinstance(assets, (Asset, ContinuousFuture, str)):
```

Comment:
Could we use `AssetConvertible` or `PricingDataAssociable` here instead, which each have the Python string types registered (and handles Python 2/3).
Checking for just `str` is not sufficient.